Bottom: bc330ad5310ef69ef66867ab688cf05709b65be1
Top:    f2f75c163c053077c440cacf9cace3dd14913e6f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-05 22:23:41 +0000

Patch out non-full-screen invalidates

(XXX: Probably breaks non-composited screen repaints horribly!)


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 88da274..f62f522 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2349,6 +2349,16 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
     object3d_test_init ();
   }
 
+//  printf ("Expose event at (%i,%i): %i x %i\n", ev->area.x, ev->area.y, ev->area.width, ev->area.height);
+
+  /* For some reason we get horrible glitching when we don't repaint the whole event region...
+   * It also appears that we get spurious small repaints interleaved with full area re-paints,
+   * so lets try just droping the small ones.
+   * NB: Currently, all our programmatically triggered repaints cover the whole region.
+   */
+  if (ev->area.x != 0 || ev->area.y != 0)
+    return; /* XXX? */
+
   gtk_widget_get_allocation (widget, &allocation);
 
   ghid_start_drawing (port, widget);
